### PR TITLE
Consolidate common code in HttpListenerRequest and fix a Managed bug

### PR DESCRIPTION
**Pull request by hughbe**
Contributes to #18128

@Priya91

### CODE REVIEW

**Comment by davidsh** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.04 Debug

**Comment by hughbe** [comment]
@dotnet-bot test OuterLoop netcoreapp Windows_NT Debug x64

File: src/System.Net.HttpListener/src/System/Net/HttpRequestStream.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+            return ReadCore(buffer, offset, size);
+        }
+
+        public override IAsyncResult BeginRead(byte[] buffer, int offset, int size, AsyncCallback callback, object state)
+        {
+            if (NetEventSource.IsEnabled)
+                NetEventSource.Enter(this);
+            if (NetEventSource.IsEnabled)
+                NetEventSource.Info(this, "buffer.Length:" + buffer.Length + " size:" + size + " offset:" + offset);
```
Comment (Priya91) [inline src/System.Net.HttpListener/src/System/Net/HttpRequestStream.cs]: We can put both of these inside single if statement..

File: src/System.Net.HttpListener/src/System/Net/HttpRequestStream.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        protected override void Dispose(bool disposing)
+        {
+            if (NetEventSource.IsEnabled)
+                NetEventSource.Enter(this);
+            try
+            {
+                if (NetEventSource.IsEnabled)
+                    NetEventSource.Info(this, "_closed:" + _closed);
+                _closed = true;
```
Comment (Priya91) [inline src/System.Net.HttpListener/src/System/Net/HttpRequestStream.cs]: This wouldn't throw, no need of try finally..

File: src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs [hunk 3]
```diff
@@ -282,6 +282,34 @@ public void Dispose()
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Read_Disposed_ReturnsZero(bool transferEncodingChunked)
+        {
```
Comment (Priya91) [inline src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs]: How about enabling the tests that differ in behavior for HttpRequestStream, this unification should have fixed some of them no?

Comment (hughbe) [inline src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs]: I don't actually think we have any tests for this before. The disabled tests are:
- Read_LargeLengthAsynchronous_Success
- Read_LargeLengthSynchronous_Success
- EndRead_InvalidAsyncResult_ThrowsArgumentException
- EndRead_CalledTwice_ThrowsInvalidOperationException
- Read_FromClosedConnectionAsynchronously_ThrowsHttpListenerException
- Read_FromClosedConnectionSynchronously_ThrowsHttpListenerException

**Comment by karelz** [comment]
@stephentoub can you please help here to push it through? Priya is OOF for next few days. Thanks!

**Pull request merged**